messagecatalog/spec.md

```diff
@@ -0,0 +1,1059 @@
+    default value is the base URI of the API endpoint.
+  - MUST be a valid URI.
+  - For Message definitions imported from other catalogs in replication
+    scenarios, the attribute is REQUIRED to be not empty. If the value is empty
+    or absent during import, it MUST be explicitly set to its implied default
+    value.
+- Examples:
+  - `urn:com-example-message definitions`
+  - `https://messagedefinitions.example.com`
+
+#### `format` (message definition)
+
+- Type: `String`
+- Description: Reflects the message definition group format. If set by a client,
+  the value MUST match the message definition group format.
+- Constraints:
+  - REQUIRED to be returned by the server. OPTIONAL for clients. IMMUTABLE.
+  - MUST be a non-empty string, if present.
+- For examples refer to the format attribute of the message definition group.
+
+#### `description` (message definition)
+
+- Type: `String`
+- Description: Explains the purpose of the message definition.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "Food order"
+
+#### `tags` (message definition group)
+
+- Type: `String`
+- Description: A single string containing a comma-separated list of tags
+  describing the message definition group for search indexing.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "exampleapp,examplecorp"
+
+#### `documentation` (message definition group)
+
+- Type: `String`
+- Description: Documents the purpose of the message definition group in greater
+  detail than the `description`. The text MAY have rich formatting in a format
+  indicated by the `documentationformat` parameter.
+- Constraints:
+  - OPTIONAL
+
+#### `documentationformat` (message definition group)
+
+- Type: `String`
+- Description: Indicates the format of the `documentation` content. Well-known
+  values are `HTML` and `Markdown` as well as `URL` if the documentation
+  attribute contains a URL pointing to external content.
+- Constraints:
+  - OPTIONAL
+
+#### `createdtimeutc` (message definition)
+
+- Type: `Timestamp`
+- Description: Instant when the Message definition was added to the catalog.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+#### `updatedtimeutc` (message definition)
+
+- Type: `Timestamp`
+- Description: Instant when the Message definition was last updated
+- Constraints:
+  - OPTIONAL
+- Assigned by the server.
+
+### 2.2.5 Message definition version
+
+A message definition version is an object that defines the concrete shape of a
+message in said version. The message definition version inherits all attributes
+of the containing message definition object.
+
+#### `version` (message definition version)
+
+- Type: `Integer`
+- Description: The version of the Message definition. This is a simple counter
+  and tracks the version in the scope of this Message definition within the
+  message definition group. The Message definition document content MAY indicate
+  a different versioning scheme.
+- Constraints:
+  - REQUIRED
+  - Assigned by server.
```

Comment:
What about push replication? Could a replication potentially result in a different version on the target?